packages/open-payments/src/client/requests.ts

```diff
@@ -0,0 +1,60 @@
+import axios, { AxiosInstance } from 'axios'
+import { ValidateFunction } from 'openapi'
+import { CreateOpenPaymentClientArgs } from '.'
+import config from '../config'
+
+export interface GetArgs {
+  url: string
+  accessToken?: string
+}
+
+export const get = async <T>(
+  axios: AxiosInstance,
+  args: GetArgs,
+  openApiResponseValidator: ValidateFunction<T>
+): Promise<T> => {
+  const { url, accessToken } = args
+
+  try {
+    const { data } = await axios.get(url, {
+      headers: accessToken
+        ? {
+            Authorization: `GNAP ${accessToken}`,
+            Signature: 'TODO',
+            'Signature-Input': 'TODO'
+          }
+        : {}
+    })
+
+    if (!openApiResponseValidator(data)) {
+      const errorMessage = 'Failed to validate OpenApi response'
+      console.log(errorMessage, {
+        url,
+        data: JSON.stringify(data)
+      })
+
+      throw new Error(errorMessage)
+    }
+
+    return data
+  } catch (error) {
+    console.log('Error when making Open Payments GET request', {
+      errorMessage: error?.message,
+      url
+    })
```

Comment:
Should this either not log or should `open-payments` support a configurable logger like `pino`?

Reply:
Yup, added pino support. wanted to do it as a follow up but it's not too big